src/commands/help.ts

```diff
@@ -0,0 +1,92 @@
+import { SlashCommandBuilder, SlashCommandStringOption } from "@discordjs/builders";
+import { RESTPostAPIApplicationCommandsJSONBody } from "discord-api-types/v9";
+import { CommandInteraction } from "discord.js";
+import { inject, injectable } from "tsyringe";
+import { c, t, useLocale } from "ttag";
+import { Command } from "../Command";
+import { COMMAND_LOCALIZATIONS, LocaleProvider } from "../locale";
+import { getLogger, Logger } from "../logger";
+import { Metrics } from "../metrics";
+
+@injectable()
+export class HelpCommand extends Command {
+	#logger = getLogger("command:help");
+
+	constructor(metrics: Metrics, @inject("LocaleProvider") private locales: LocaleProvider) {
+		super(metrics);
+	}
+
+	static override get meta(): RESTPostAPIApplicationCommandsJSONBody {
+		const builder = new SlashCommandBuilder().setName("help").setDescription("Get help with a Slash Command.");
+
+		// TODO: un-hardcode this? would require wrestling with circular deps, and would probably still need
+		// manual exclusions in case of undocumented commands like ping
+		const documentedCommands = [
+			"art",
+			"deck",
+			"id",
+			"link",
+			"locale",
+			"random",
+			"search",
+			"ygoprodeck",
+			"yugipedia"
+		];
+
+		const option = new SlashCommandStringOption()
+			.setName("command")
+			.setDescription("The name of a specific Slash Command.")
+			.addChoices(
+				...documentedCommands.map(s => {
+					return { name: s, value: s };
+				})
+			)
+			.setRequired(false);
+
+		for (const { gettext, discord } of COMMAND_LOCALIZATIONS) {
+			useLocale(gettext);
+			builder
+				.setNameLocalization(discord, c("command-name").t`help`)
+				.setDescriptionLocalization(discord, c("command-description").t`Get help with a Slash Command.`);
+			option
+				.setNameLocalization(discord, c("command-option").t`command`)
+				.setDescriptionLocalization(
+					discord,
+					c("command-option-description").t`The name of a specific Slash Command.`
+				);
+		}
+
+		builder.addStringOption(option);
+
+		return builder.toJSON();
+	}
+
+	protected override get logger(): Logger {
+		return this.#logger;
+	}
+
+	protected override async execute(interaction: CommandInteraction): Promise<number> {
+		const command = interaction.options.getString("command", false);
+		const resultLanguage = await this.locales.get(interaction);
+
+		let output: string;
+
+		if (command) {
+			output = t`Documentation for the \`/${command}\` Slash Command:\n<https://github.com/DawnbrandBots/bastion-bot/blob/master/docs/commands/${command}.md>`;
+		} else {
+			output = t`Documentation for Slash Commands:\n<https://github.com/DawnbrandBots/bastion-bot/tree/master/docs/commands>`;
+		}
```

Comment:
I think it's good enough to link. We still have yet to see what localising the command names looks like in Discord's UI and whether that's any good. ~Also then we don't need to translate text here~